core/citrus-api/src/main/java/org/citrusframework/spi/ResourcePathTypeResolver.java

```diff
@@ -232,25 +267,32 @@ private Constructor<?> getConstructor(Class<?> type, Object[] initargs) {
 
     /**
      * Read resource from classpath and load content as properties.
+     * The properties found on the classpath will be cached.
+     *
      * @param resourcePath
      * @return
      */
     private Properties readAsProperties(String resourcePath) {
-        String path = getFullResourcePath(resourcePath);
-
-        InputStream in = ResourcePathTypeResolver.class.getClassLoader().getResourceAsStream(path);
-        if (in == null) {
-            throw new CitrusRuntimeException(String.format("Failed to locate resource path '%s'", path));
-        }
+        return resourceProperties.computeIfAbsent(resourcePath, k -> {
+            String path = getFullResourcePath(resourcePath);
+
+            InputStream in = ResourcePathTypeResolver.class.getClassLoader()
```

Comment:
I find it very hard to read, to be honest. its' probably IntelliJ auto-formatting and I know we have no `prettier` or any other rules in this project.

but, while looking through it I did at least find this line: this results in open handles.

I will follow up with another PR, reformatting this (and any other issues I might find).

Reply:
see 3443c8bb94f9c181d0041f8afaed43c73aec2697 in https://github.com/citrusframework/citrus/pull/1044.